### ISSUE TITLE
Undetermined start location on FB should ALWAYS default to city of kennel not a random spot on the planet

### ISSUE THREAD

**Issue by James-A-White** [James-A-White/HarrierCentral#134]
Undetermined
Unknown
TBD
To be determined
TBA
To be announced

**Comment by diabWH3** [comment]
Good idea. I've found that "TBD" can actually geolocate as a POI when using
Google Maps.

It causes issues with tools like my Seattle-area heatmap: pulls the last
365 days of calendar info for our hash website and adds pins to a map
(mostly trail locations, but some happy hour spots too).
I have repeating placeholder calendar events that all used to have "TBD" as
the Location field and those placeholders would go on the map in unexpected
places.
https://wh3.org/harelines/heatmap/

On-On,

*Dick in a Box*

**Comment by James-A-White** [comment]
This is fixed now for runs that are imported from Facebook. If a user manual enters a run, we don’t try to resolve the address because we assume they’ve use the map to put a location pin in the correct spot.

If you think it’s important for us to try to resolve the location for runs that are manually entered into the system. Please open a new issue.

**Comment by James-A-White** [comment]
I’ve made other improvements to the Geo resolving as well. First of all if a kennel doesn’t have its own lat long we assume the lat long of the city. This was not working properly before. Second, we’re trying to resolve a Geo location from Free text, I constrain the search to the country where the kennel is located and any neighboring countries. This is not so important in the United States, but for countries like the Netherlands were run could easily be in Belgium, it makes sense to enlarge the search field to include neighboring countries. And finally I’ve added logic to look at the quality of the search results And provide a bias for search results that have more complete information such as the street address ZIP Code the name of a location such as a restaurant, etc.

**Comment by James-A-White** [comment]
At this point I think we should let it run for a month or two and see what the quality the results looks like. I can further tweak it if required. If worse comes to worse I can select another Geo location service such as from Google, but I’m trying to stay in the azure ecosystem as much as possible.

**Comment by James-A-White** [comment]
One final note, if a free text location is injured and Facebook that does not correspond to a Facebook place, if that text is less than or equal to three characters long, I will not try to resolve the location and The system will store a “no location “flag so that nothing is displayed on the map.
